client/js/services/articles.service.js

```diff
@@ -124,25 +123,23 @@
                     if (feed.format === "RSS") {
                         items = xmlDoc.getElementsByTagName('item');
                         for (var i = 0; i < items.length; i++) {
-                            var articleObj = {
-                                title: items[i].getElementsByTagName('title')[0].innerHTML,
-                                link: items[i].getElementsByTagName('link')[0].textContent,
-                                img: getImage(items[i], feed.format),
-                                content: getContent(items[i], feed.format) || this.title,
-                                date: items[i].getElementsByTagName('pubDate')[0].textContent
-                            };
+                            var articleObj = {};
+                            articleObj.title = items[i].getElementsByTagName('title')[0].innerHTML;
+                            articleObj.link = items[i].getElementsByTagName('link')[0].textContent;
+                            articleObj.img = getImage(items[i], feed.format);
+                            articleObj.content = getContent(items[i], feed.format) || articleObj.title;
+                            articleObj.date = items[i].getElementsByTagName('pubDate')[0].textContent;
                             obj.articles.push(articleObj);
                         }
                     } else if (feed.format === "ATOM") {
                         items = xmlDoc.getElementsByTagName('entry');
                         for (var i = 0; i < items.length; i++) {
-                            var articleObj = {
-                                title: items[i].getElementsByTagName('title')[0].textContent,
-                                link: items[i].getElementsByTagName('id')[0].textContent,
-                                img: getImage(items[i], feed.format),
-                                content: getContent(items[i], feed.format) || this.title,
-                                date: items[i].getElementsByTagName('published')[0].textContent
-                            };
+                            var articleObj = {};
```

Comment:
If i choose second approach, i get undefined values for some of this properties.